### PR TITLE
[Icons] Refactored Figma fetch to TS

### PR DESCRIPTION
**Pull request by KenAJoh**
### Description

Code was causing some linting issues and notices some potential bugs not picked up since it was just JS. Functionality is the same, but now with types and better error-handling

### Component Checklist 📝

- [x] JSDoc
- [x] Examples
- [x] Documentation
- [x] Storybook
- [x] Style mappings (`@navikt/core/css/config/_mappings.js`)
- [x] Component tokens (`@navikt/core/css/tokens.json`)
- [x] CSS class deprecations (`@navikt/aksel-stylelint/src/deprecations.ts`)
- [x] Exports (`@navikt/core/react/src/index.ts` and `@navikt/core/react/package.json`)
- [x] New component? CSS import (`@navikt/core/css/index.css`)
- [x] Breaking change? Update migration guide. Consider codemod.
- [x] Changeset (Format: `<Component>: <gitmoji?> <Text>.` E.g. "Button: :sparkles: Add feature xyz.")

### CODE REVIEW

File: @navikt/aksel-icons/config/figma/fetch-icons.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    }
+  }
+  return idUrlPairs;
+};
+
+/**
+ * Splits an array of strings into parts where each part is a comma-separated string of all the strings for that part.
+ * @returns An array of comma-separated strings.
+ */
+export function splitIntoParts(arr: string[], pages: number): string[] {
```
Comment (HalvorHaugan) [inline @navikt/aksel-icons/config/figma/fetch-icons.ts]: `pages` -> `parts`?

File: @navikt/aksel-icons/config/figma/index.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        throw e.message;
+      });
+
+    if (!icon) {
+      continue;
+    }
+
+    /*
+     * Arbitrary delay to not get rate-limited by image hosting
+     * Currently accounts for ~18 seconds in theory, but in practice the bottlenech is fetching each icon 1 at a time
```
Comment (HalvorHaugan) [inline @navikt/aksel-icons/config/figma/index.ts:49]:
```suggestion
     * Currently accounts for ~18 seconds in theory, but in practice the bottleneck is fetching each icon 1 at a time
```

File: @navikt/aksel-icons/config/figma/index.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  throw new Error("FIGMA_TOKEN not set in .env");
+}
+
+main();
+
+async function main() {
+  console.info("Started icon-update from Figma");
+  const icons = await fetchIcons();
+
+  const images = await fetchDownloadUrls(icons.map((x) => x.node_id));
```
Comment (HalvorHaugan) [inline @navikt/aksel-icons/config/figma/index.ts]: `images` -> `imageUrls`?

File: @navikt/aksel-icons/config/figma/index.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  if (existsSync(iconFolder)) {
+    rmSync(iconFolder, { recursive: true, force: true });
+  }
+
+  mkdirSync(iconFolder);
+
+  console.info(`Downloading ${Object.keys(images).length} icons from Figma...`);
+
+  let counter = 0;
+  for (const [key, value] of Object.entries(images)) {
```
Comment (HalvorHaugan) [inline @navikt/aksel-icons/config/figma/index.ts]: Can we find some more describing names for `key` and `value`? Maybe "iconId" and "iconUrl"?

File: @navikt/aksel-icons/config/figma/index.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+if (!process.env.FIGMA_TOKEN) {
+  throw new Error("FIGMA_TOKEN not set in .env");
+}
+
+main();
+
+async function main() {
+  console.info("Started icon-update from Figma");
+  const icons = await fetchIcons();
```
Comment (HalvorHaugan) [inline @navikt/aksel-icons/config/figma/index.ts]: Would "iconsMetadata" be more describing than just "icons"?

File: @navikt/aksel-icons/config/figma/index.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    rmSync(iconFolder, { recursive: true, force: true });
+  }
+
+  mkdirSync(iconFolder);
+
+  console.info(`Downloading ${Object.keys(images).length} icons from Figma...`);
+
+  let counter = 0;
+  for (const [key, value] of Object.entries(images)) {
+    const icon = await fetch(value)
```
Comment (HalvorHaugan) [inline @navikt/aksel-icons/config/figma/index.ts]: `icon` -> `iconSvg`?

File: @navikt/aksel-icons/config/figma/index.ts [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+  }
+  console.info(`Completed processing of ${counter} icons`);
+
+  makeConfig(icons, iconFolder);
+
+  const filesInDir = readdirSync(iconFolder);
+
+  if (filesInDir.length * 2 !== icons.length) {
+    throw new Error(
+      `Icons written to director (${filesInDir.length}) does not match the amount of icons located in Figma (${icons.length})`,
```
Comment (HalvorHaugan) [inline @navikt/aksel-icons/config/figma/index.ts]:
```suggestion
      `Icons written to directory (${filesInDir.length}) does not match the amount of icons located in Figma (${icons.length})`,
```

File: @navikt/aksel-icons/config/figma/make-configs.ts [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    if (!icon.containing_frame?.pageName || !icon.containing_frame?.name) {
+      throw new Error(
+        `Missing containing_frame.pageName or containing_frame.name for icon ${name}`,
+      );
+    }
+
+    const config: IconYml = {
+      name,
+      category: icon.containing_frame?.pageName,
+      sub_category: icon.containing_frame?.name,
```
Comment (HalvorHaugan) [inline @navikt/aksel-icons/config/figma/make-configs.ts]:
```suggestion
      category: icon.containing_frame.pageName,
      sub_category: icon.containing_frame.name,
```

File: @navikt/aksel-icons/config/figma/index.ts (hunk 5 above)
Comment (KenAJoh) [reply @navikt/aksel-icons/config/figma/index.ts]: Went for `publishedIconComponents`

**Review by HalvorHaugan — APPROVED** [review]

**Pull request merged**